partiql-test-data/fail/parser/primitives/date-constructor.ion

```diff
@@ -0,0 +1,183 @@
+    statement: "DATE 'date_string'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string format missing dashes",
+    statement: "DATE '20210310'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string format unexpected colons",
+    statement: "DATE '2021:03:10'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string no starting single quotation",
+    statement: "DATE 2021-03-10'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string no ending single quotation",
+    statement: "DATE '2021-03-10",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string format MMDDYYYY",
+    statement: "DATE '03-25-2021'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string format DDMMYYYY",
+    statement: "DATE '25-03-2021'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string minus before year",
+    statement: "DATE '-9999-03-10'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string plus before year",
+    statement: "DATE '+9999-03-10'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string extra minus between year and month",
+    statement: "DATE '2021--03-10'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string plus between year and month",
+    statement: "DATE '2021-+03-10'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string extra minus between month and day",
+    statement: "DATE '2021-03--10'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string plus between month and day",
+    statement: "DATE '2021-03-+10'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string month out of range",
+    statement: "DATE '2021-12345-10'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string day out of range",
+    statement: "DATE '2021-03-12345'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string year missing padded zero",
+    statement: "DATE '123-03-10'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string month missing padded zero",
+    statement: "DATE '2021-3-10'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string day missing padded zero",
+    statement: "DATE '2021-03-1'",
+    assert: {
+        result: ParseError
+    },
+}
+
+parse::{
+    name: "invalid DATE string year beyond 9999",
+    statement: "DATE '12345-03-10'",
+    assert: {
+        result: ParseError
+    },
+}
```

Comment:
The SQL-92 spec says the year is 4 digits while month and day are both 2 digits:

```
Within a <datetime literal>, the <years value> shall contain
            four digits. The <seconds integer value> and other datetime
            components, with the exception of <seconds fraction>, shall each
            contain two digits.
```

PostgreSQL allows these other `DATE`s (e.g. missing leading zero or year with more than 4 digis).

Reply:
I agree, let's leave it at SQL Compatibility for now.